internal/receiver/discoveryreceiver/README.md

```diff
@@ -0,0 +1,334 @@
+Severity: debug
+Body: Container appears to not be accepting redis connections.
+Attributes:
+     -> name: STRING(smartagent/redis/receiver_creator/discovery{endpoint="172.17.0.2:80"}(54141aa1da2d2ad0))
+     -> monitorType: STRING(collectd/redis)
+     -> logger: STRING(root)
+     -> createdTime: STRING(1.6589422177182763e+09)
+     -> level: STRING(error)
+     -> monitorID: STRING(smartagentredisreceiver_creatordiscoveryendpoint17217028054141aa1da2d2ad0)
+     -> runnerPID: STRING(119590)
+     -> sourcePath: STRING(/usr/lib/splunk-otel-collector/agent-bundle/collectd-python/redis/redis_info.py)
+     -> lineno: STRING(198)
+     -> caller: STRING(signalfx/handler.go:189)
+     -> receiver.name: STRING(smartagent/redis)
+     -> discovery.status: STRING(failed)
+Trace ID:
+Span ID:
+Flags: 0
+```
+
+## Config
+
+### Main
+
+| Name | Type | Default | Docs |
+| ---- | ---- | ------- | ---- |
+| `watch_observers` (required) | []string | <no value> | The array of Observer extensions to receive Endpoint events from |
+| `log_endpoints` | bool | false | Whether to emit log records for Observer Endpoint events |
+| `embed_receiver_config` | bool | false | Whether to embed a base64-encoded, minimal Receiver Creator config for the generated receiver as a reported metrics `discovery.receiver.rule` resource attribute value for status log record matches |
+| `receivers` | map[string]ReceiverConfig | <no value> | The mapping of receiver names to their Receiver sub-config |
+
+### ReceiverConfig
+
+| Name | Type | Default | Docs |
+| ---- | ---- | ------- | ---- |
+| `rule` (required) | string | <no value> | The Receiver Creator compatible discover rule |
+| `config` | map[string]interface{} | <no value> | The receiver instance configuration, including any Receiver Creator endpoint env value expr program value expansion |
+| `resource_attributes` | map[string]string | <no value> | A mapping of string resource attributes and their (expr program compatible) values to include in reported metrics for status log record matches |
+| `status` | map[string]Match | <no value> | A mapping of `metrics` and/or `statements` to Match items for status evaluation |
+
+### Match
+
+**One of `regexp`, `strict`, or `expr` is required.**
+
+| Name | Type | Default | Docs |
+| ---- | ---- | ------- | ---- |
+| `regexp` | string | <no value> | The regexp pattern to evaluate reported received metric names or component log statements |
+| `strict` | string | <no value> | The string literal to compare equivalence against reported received metric names or component log statements |
+| `expr` | string | <no value> | The expr program run with the reported received metric names or component log statements (vm env TBD) |
+| `first_only` | bool | false | Whether to emit only one log record for the first matching metric or log statement, ignoring all subsequent matches |
+| `record` | LogRecord | <no value> | The emitted log record content |
+
+### LogRecord
```

Comment:
Are these the log records that could be consumed by LO and then visualized on data setup views? 

Thinking aloud, (this is not specific to this PR; moreover a question we on the product side need to answer), if a customer does not have a LO entitlement (review inventory on the data setup wizard only visualizes logs if they do), what is the best way for us to guide the customer for visibility?

Reply:
This seems applicable to log records from the collector in general and without LO. One way would be to establish practices for using attributes or transform processor to insert the required record attributes to convert them to metric events (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/splunk/common.go#L27, https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8e4a062fa9aef1543b477a4b3469085d4b1c95db/exporter/signalfxexporter/internal/translation/logdata_to_signalfxv2.go#L49). The other would be to make the discovery receiver a traces pipeline compatible component and emit traces for each endpoint event with endpoints as spans.

Reply:
Great ideas! I'll set up some time to chat about our direction moving forward based on these recommendations.

Reply:
I believe we should use SignalFx metric events since the discovered receivers are mostly for metrics. APM also can be disabled on Splunk O11y side